src/Sarif.Converters/StaticDriverVerifierConverter.cs

```diff
@@ -162,29 +171,57 @@ private void ProcessLine(string logFileLine, Result result)
-                // no location. This actually code cause problems, as a some
-                // steps with locations are embedded in call/returns that 
-                // are discarded. This converter (and the viewer experience)
-                // should be compared soon with the actual SDV viewer, with
-                // a revisit of the converter's behavior.           
+                var state = tokens[STATE];
+                string[] stateTokens = state.Split(new string[] { "^====Auto=====" }, StringSplitOptions.RemoveEmptyEntries);
 
+                if (stateTokens.Length == 1)
+                {
+                    // We have a single token only because no data values were specified
+                    if (state.StartsWith("^====Auto====="))
+                    {
+                        annotatedCodeLocation.SetProperty("permVars", stateTokens[0]);
+                    }
+                    else
+                    {
+                        // Only data values were specified, no permanent vars
+                        annotatedCodeLocation.SetProperty("dataValuesCurrent", stateTokens[0]);
+                    }
+                }
+                else if (stateTokens.Length > 1)
+                {
+                    annotatedCodeLocation.SetProperty("dataValuesCurrent", stateTokens[0]);
+                    annotatedCodeLocation.SetProperty("permVars", stateTokens[1]);
+                }
 
-                if (sdvKind.Equals("Call", StringComparison.Ordinal))
+                if (kind == AnnotatedCodeLocationKind.Call)
                 {
-                    // Caller is encapsulated in double quotes, which we trim
-                    annotatedCodeLocation.SetProperty("Caller", tokens[CALLER].Trim('"'));
+                    codeLocationStack.Push(annotatedCodeLocation);
+
+                    string[] callTokens = logFileLine.Split(new string[] { " Call " }, StringSplitOptions.RemoveEmptyEntries);
+                    Debug.Assert(callTokens.Length == 2);
 
-                    // Callee is at the end of the line. First we trim the \r\b.
-                    // Next remove the encapsulating double-quotes.
-                    annotatedCodeLocation.SetProperty("Callee", tokens[CALLEE].Trim().Trim('"'));
+                    string extraMsg = "Call " + callTokens[1].Replace('"', '\'').Trim();
+                    annotatedCodeLocation.SetProperty("extraMsg", extraMsg);
+
+                    if (IsImportantCall(uriValue, extraMsg))
+                    {
+                        foreach(AnnotatedCodeLocation acl in codeLocationStack)
+                        {
+                            if (acl.Importance == AnnotatedCodeLocationImportance.Unimportant)
+                            {
+                                acl.Importance = AnnotatedCodeLocationImportance.Essential;
+                            }
+                        }
+                    }
```

Comment:
Oh wow. Lightbulb. It took me reading up to here before I understood why you were keeping the stack -- it's so that if you find a deeply nested call that you consider important, you will light up all the callers as well. This is worth a comment.